packages/maker.js/src/core/model.ts

```diff
@@ -111,17 +111,17 @@ namespace MakerJs.model {
      * @returns Array of Caption objects.
      */
     export function getAllCaptionsOffset(modelContext: IModel) {
-        const captions: ICaption[] = [];
+        const captions: (ICaption & { parentLayer?: string })[] = [];
```

Comment:
Perhaps `parentLayer` naming is a bit too specific? Since it is a cascaded layer, it might be a "great great grandparent".